### PR TITLE
Pipeline: complete method returns a bool to indicate if complete is successful

### PR DESCRIPTION
**Pull request by cyanglaz**
Add a return type "bool" so we know if complete is successful.
Also update the rasterizer in re-submit scenario to make the `consume_result` to be `PipelineConsumeResult::MoreAvailable;` if the `complete` is successful during re-submit.

Updated the test to check the result.

### CODE REVIEW

File: shell/common/pipeline.h [hunk 1]
```diff
@@ -60,20 +60,22 @@ class Pipeline : public fml::RefCountedThreadSafe<Pipeline<R>> {
       }
     }
 
-    void Complete(ResourcePtr resource) {
+    bool Complete(ResourcePtr resource) {
```
Comment (chinmaygarde) [inline shell/common/pipeline.h:63]: These seem like a good candidates to decorate with `FML_WARN_UNUSED_RESULT`.

File: shell/common/rasterizer.cc [hunk 2]
```diff
@@ -131,7 +131,14 @@ void Rasterizer::Draw(fml::RefPtr<Pipeline<flutter::LayerTree>> pipeline) {
   // front of the queue and also change the consume status to more available.
   if (raster_status == RasterStatus::kResubmit) {
     auto front_continuation = pipeline->ProduceIfEmpty();
-    front_continuation.Complete(std::move(resubmitted_layer_tree_));
+    if (front_continuation) {
```
Comment (chinmaygarde) [inline shell/common/rasterizer.cc:134]: Can we separate this into another patch please? This bit of the patch does not line up with the changes stated in the commit message.

File: shell/common/pipeline.h (hunk 1 above)
Comment (cyanglaz) [reply shell/common/pipeline.h:63]: Will do!

Comment (cyanglaz) [reply shell/common/pipeline.h:63]: FML_WARN_UNUSED_RESULT doesn't seem to exist anymore. Removed in https://github.com/flutter/engine/commit/a37560a07447befc74e5c418f00cac0cad4e1404
Should we bring it back? Maybe in a seperate PR?

Comment (chinmaygarde) [reply shell/common/pipeline.h:63]: Sorry, I meant [[nodiscard]] which is the C++17 way of specifying `FML_WARN_UNUSED_RESULT`.

Comment (cyanglaz) [reply shell/common/pipeline.h]: done!

File: shell/common/animator.cc [hunk 3]
```diff
@@ -185,7 +185,10 @@ void Animator::Render(std::unique_ptr<flutter::LayerTree> layer_tree) {
   }
 
   // Commit the pending continuation.
-  producer_continuation_.Complete(std::move(layer_tree));
+  bool result = producer_continuation_.Complete(std::move(layer_tree));
+  if (!result) {
+    FML_DLOG(INFO) << "No pending continuation to commit";
```
Comment (cyanglaz) [inline shell/common/animator.cc:190]: I was considering `FML_DCHECK` the result, but some shell_tests fails the check.
I suppose it is expected that `Complete` fails here? Or do we always expect the `Complete` to success here? (meaning the `continuation_` is never null here.) 
@chinmaygarde

Comment (chinmaygarde) [reply shell/common/animator.cc:190]: I am not able to reason about when that would fail and if on failure, the delegates OnAnimatorDraw invocation needs to be made. cc @iskakaushik 

I suppose it is fine to log for now as this was not the point of your change.
